lib/modules/hosts/deviantart.js

```diff
@@ -13,12 +13,13 @@ addLibrary('mediaHosts', 'deviantart', {
 			url: apiURL,
 			// aggressiveCache: true,
 			onload: function(response) {
+				var json;
 				try {
-					var json = JSON.parse(response.responseText);
-					def.resolve(elem, json);
+					json = JSON.parse(response.responseText);
```

Comment:
Just in case it's not clear, resolve here was calling the handleInfo method, so exceptions thrown from that method were being caught by this try catch block here.